peer.go

```diff
@@ -1531,6 +1543,9 @@ out:
 			p.handleTxMsg(msg)
 
 		case *wire.MsgBlock:
+			if p.blockStallCancel != nil {
```

Comment:
This is reading the peer field that is being written in `outHandler` which is a separate goroutine.  So it's a race.  The race detect confirms:

==================
WARNING: DATA RACE
Read by goroutine 55:
  main.(*peer).inHandler()
      peer.go:1546 +0x147b

Previous write by goroutine 57:
  main.(*peer).outHandler()
      peer.go:1835 +0x12b1
==================